fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkexport/patient/ChunkReader.java

```diff
@@ -108,73 +126,88 @@ public ChunkReader() {
+                        searchContext.setPageNumber(compartmentPageNum);
+                        FHIRTransactionHelper txn = new FHIRTransactionHelper(fhirPersistence.getTransaction());
+                        txn.enroll();;
+                        FHIRPersistenceContext persistenceContext = FHIRPersistenceContextFactory.createPersistenceContext(null, searchContext);
+                        List<Resource> resources = fhirPersistence.search(persistenceContext, resourceType).getResource();
+                        txn.unenroll();;
+                        compartmentPageNum++;
+
+                        for (Resource res : resources) {
+                            if (res == null || (isDoDuplicationCheck && loadedResourceIds.contains(res.getId()))) {
+                                continue;
+                            }
+                            try {
+                                FHIRGenerator.generator(Format.JSON).generate(res, chunkData.getBufferStream());
+                                chunkData.getBufferStream().write(Constants.NDJSON_LINESEPERATOR);
+                                resSubTotal++;
+                                if (isDoDuplicationCheck) {
+                                    loadedResourceIds.add(res.getId());
+                                }
+                            } catch (FHIRGeneratorException e) {
+                                if (res.getId() != null) {
+                                    logger.log(Level.WARNING, "fillChunkDataBuffer: Error while writing resources with id '"
+                                            + res.getId() + "'", e);
+                                } else {
+                                    logger.log(Level.WARNING,
+                                            "fillChunkDataBuffer: Error while writing resources with unknown id", e);
+                                }
+                            } catch (IOException e) {
+                                logger.warning("fillChunkDataBuffer: chunkDataBuffer written error!");
+                                throw e;
                             }
-                        } catch (IOException e) {
-                            logger.warning("fillChunkDataBuffer: chunkDataBuffer written error!");
-                            throw e;
                         }
-                    }
 
-                } while (searchContext.getLastPageNumber() >= compartmentPageNum);
-            }
+                    } while (searchContext.getLastPageNumber() >= compartmentPageNum);
+                }
+
+                indexOfCurrentTypeFilter++;
+            } while (searchParametersForResoureTypes.get(resourceType) != null && indexOfCurrentTypeFilter < searchParametersForResoureTypes.get(resourceType).size());
+
             chunkData.setCurrentPartResourceNum(chunkData.getCurrentPartResourceNum() + resSubTotal);
             logger.fine("fillChunkDataBuffer: Processed resources - " + resSubTotal + "; Bufferred data size - "
```

Comment:
this didn't occur to me earlier, maybe wrap this in isLogLevel

Reply:
sounds good, let me change it.

Reply:
done